### PR TITLE
add  RegistryCronJob api and  UpdatedTimeField

### PR DESCRIPTION
**Pull request by cigra**

### CODE REVIEW

File: service_test.go [hunk 1]
```diff
@@ -420,8 +420,52 @@ func (suite *MainTestSuite) TestFrameworks() {
...
+var rCmpFilter = cmp.FilterPath(func(p cmp.Path) bool {
+	return p.String() == "PortalBase.GUID" || p.String() == "CreationTime" || p.String() == "PortalBase.UpdatedTime"
+}, cmp.Ignore())
+
+func (suite *MainTestSuite) TestRegistryCronJobs() {
+	registryCronJobs, _ := loadJson[*types.RegistryCronJob](registryCronJobJson)
+
+	modifyFunc := func(r *types.RegistryCronJob) *types.RegistryCronJob {
+		if r.Include == nil {
+			r.Include = []string{}
```
Comment (avrahams) [inline service_test.go:441]: Do we want "include" to be searchable in the queries 
if so we need to add it to the config and mark it as array = true

File: service_test.go [hunk 2]
```diff
@@ -420,8 +420,52 @@ func (suite *MainTestSuite) TestFrameworks() {
...
+		{
+			query:           "clusterName=minikube&registryName=gc",
+			expectedIndexes: []int{2},
+		},
+	}
+
+	testGetDeleteByNameAndQuery(suite, consts.RegistryCronJobPath, consts.NameField, registryCronJobs, getQueries)
+
+	r1 := testPostDoc(suite, consts.RegistryCronJobPath, registryCronJobs[0], rCmpFilter)
+	updateTime, err := time.Parse(time.RFC3339, r1.UpdatedTime)
```
Comment (avrahams) [inline service_test.go:466]: Very good that we test it 
Next time we will do work here let's add GetUpdateTime() to portal base and add this test to the common generic test for all elements

**Review by avrahams — CHANGES_REQUESTED** [review]
Looks very good !
Few more comments:
1. add your test data to join the test in  ```func (suite *MainTestSuite) TestAdminAndUsers() ``` like the other document types 
2. update the test command line in README (L 208) with your new package 
3. Run the coverage report and make sure the code is covered (especially ```CustomerConfig``` update time in  in ```types.go ```)
Check the coverage percentage - it shouldn't drop less than 75%

File: service_test.go (hunk 2 above)
Comment (cigra) [reply service_test.go:466]: UpdatedTime already in  portal base, I'll move check to common test

File: service_test.go (hunk 1 above)
Comment (cigra) [reply service_test.go:441]: not need now

File: service_test.go [hunk 3]
```diff
@@ -420,8 +420,52 @@ func (suite *MainTestSuite) TestFrameworks() {
...
+	modifyFunc := func(r *types.RegistryCronJob) *types.RegistryCronJob {
+		if r.Include == nil {
+			r.Include = []string{}
+		}
+		r.Include = append(r.Include, "new-registry"+rndStr.NewLen(5))
+		return r
+	}
+	commonTest(suite, consts.RegistryCronJobPath, registryCronJobs, modifyFunc, rCmpFilter)
+
+	getQueries := []queryTest[*types.RegistryCronJob]{
```
[the remark this answers is not in the document]
Comment (cigra) [reply service_test.go:449]: done, without authMethod.type(not relevant now)

**Comment by cigra** [comment]
> Looks very good ! Few more comments:
> 
> 1. add your test data to join the test in  `func (suite *MainTestSuite) TestAdminAndUsers() ` like the other document types
> 2. update the test command line in README (L 208) with your new package
> 3. Run the coverage report and make sure the code is covered (especially `CustomerConfig` update time in  in `types.go `)
>    Check the coverage percentage - it shouldn't drop less than 75%

done, coverage 76.1% (100% for RegistryCronJob)

**Review by avrahams — APPROVED** [review]

**Pull request merged**